clang/runtime/dpct-rt/include/dnnl_utils.hpp

```diff
@@ -2489,6 +2626,9 @@ sycl::event engine_ext::execute_primitive(
     }
   }
   caches.insert(caches.end(), device_ptrs.begin(), device_ptrs.end());
+  if(preserve_primitive) {
+    primitive = nullptr;
```

Comment:
Please refine the code: as it seems the memory pointer by primitive will never be freeed.